File: android/app/src/main/java/com/laneshadow/ui/components/molecules/AuthCard.kt
package com.laneshadow.ui.components.molecules

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.laneshadow.theme.LocalLaneShadowTheme

/**
 * AuthCard molecule component
 *
 * Card container for authentication screens (login, signup).
 * Provides a themed card with optional icon, title, subtitle, and content area.
 *
 * @param title Card title text
 * @param subtitle Optional subtitle/description text
 * @param icon Optional top icon composable
 * @param content Card body content composable
 * @param modifier Modifier for the card container
 */
@Composable
fun AuthCard(
    title: String,
    subtitle: String? = null,
    icon: @Composable (() -> Unit)? = null,
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit,
) {
    val theme = LocalLaneShadowTheme.current

    // Card styling constants
    val cornerRadius = theme.radius.xl
    val cardPadding = theme.space.lg
    val elevation = 1.dp
    val borderStroke = BorderStroke(1.dp, theme.colors.border.default)

    Surface(
        modifier = modifier,
        shape = androidx.compose.foundation.shape.RoundedCornerShape(cornerRadius),
        color = theme.colors.surface.default,
        border = borderStroke,
        tonalElevation = elevation,
    ) {
        Column(
            modifier = Modifier
                .padding(cardPadding),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            // Optional icon at top
            if (icon != null) {
                Box(
                    modifier = Modifier
                        .padding(bottom = 24.dp),
                    contentAlignment = Alignment.Center,
                ) {
                    icon()
                }
            }

            // Title
            Text(
                text = title,
                modifier = Modifier.padding(bottom = if (subtitle != null) 8.dp else 24.dp),
                style = androidx.compose.ui.text.TextStyle(
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                ),
                color = theme.colors.onSurface.default,
            )

            // Optional subtitle
            if (subtitle != null) {
                Text(
                    text = subtitle,
                    modifier = Modifier.padding(bottom = 24.dp),
                    style = androidx.compose.ui.text.TextStyle(
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Normal,
                    ),
                    color = theme.colors.onSurface.default.copy(alpha = 0.7f),
                )
            }

            // Card content
            content()
        }
    }
}

File: ios/LaneShadow/Views/Molecules/AuthCard.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Auth Card Component

/**
 * Auth card molecule component
 *
 * Card container for authentication flows with optional icon, title, subtitle, and content.
 * Following the React Native component from react-native/components/ui/molecules/auth-card.tsx
 *
 * ## Design Tokens Used
 * - Colors:
 *   - `theme.colors.surface.default` (background)
 *   - `theme.colors.border.default` (border)
 *   - `theme.colors.onSurface.default` (title/subtitle text)
 * - Layout:
 *   - Padding: 16pt
 *   - Border radius: 16pt
 *   - Border width: 1pt
 * - Typography:
 *   - Title: 16pt semibold
 *   - Subtitle: 14pt regular with 0.7 opacity
 * - Elevation:
 *   - Shadow radius 2pt, y offset 1pt
 *
 * ## Parameters
 * - title: Card title text
 * - subtitle: Optional subtitle/description text
 * - icon: Optional top icon view
 * - content: Card body content via ViewBuilder
 */
public struct LSAuthCard<Content: View>: View {
    @Environment(\.theme) private var theme

    private let title: String
    private let subtitle: String?
    private let icon: AnyView?
    private let content: () -> Content

    public init(
        title: String,
        subtitle: String? = nil,
        icon: AnyView? = nil,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.title = title
        self.subtitle = subtitle
        self.icon = icon
        self.content = content
    }

    // MARK: - Body

    public var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            // Icon area
            if let icon {
                HStack {
                    Spacer()
                    icon
                    Spacer()
                }
            }

            // Title and subtitle
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundStyle(theme.colors.onSurface.default)

                if let subtitle {
                    Text(subtitle)
                        .font(.system(size: 14, weight: .regular))
                        .foregroundStyle(theme.colors.onSurface.default.opacity(0.7))
                }
            }

            // Content area
            content()
        }
        .frame(maxWidth: .infinity)
        .padding(16)
        .background(theme.colors.surface.default)
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(theme.colors.border.default, lineWidth: 1)
        )
        .shadow(
            color: Color.black.opacity(0.1),
            radius: 2,
            y: 1
        )
    }
}

// MARK: - Preview

#Preview("AuthCard - Basic") {
    LSAuthCard(title: "Welcome Back") {
        Text("Sign in to continue")
            .font(.body)
    }
    .laneShadowTheme()
    .padding()
}

#Preview("AuthCard - With Subtitle") {
    LSAuthCard(
        title: "Welcome Back",
        subtitle: "Enter your credentials to access your account"
    ) {
        Text("Sign in form goes here")
            .font(.body)
    }
    .laneShadowTheme()
    .padding()
}

#Preview("AuthCard - With Icon") {
    LSAuthCard(
        title: "Create Account",
        subtitle: "Join thousands of riders",
        icon: AnyView(
            Image(systemName: "person.circle.fill")
                .font(.system(size: 48))
                .foregroundStyle(Color.blue)
        )
    ) {
        Text("Sign up form goes here")
            .font(.body)
    }
    .laneShadowTheme()
    .padding()
}

#Preview("AuthCard - Complete") {
    LSAuthCard(
        title: "Sign In",
        subtitle: "Welcome back to LaneShadow",
        icon: AnyView(
            Image(systemName: "motorcycle")
                .font(.system(size: 48))
                .foregroundStyle(Color.blue)
        )
    ) {
        VStack(spacing: 12) {
            Text("Email")
                .font(.caption)
                .foregroundStyle(.secondary)
            Text("user@example.com")
                .font(.body)
                .padding(.vertical, 8)
                .padding(.horizontal, 12)
                .frame(maxWidth: .infinity)
                .background(Color.gray.opacity(0.1))
                .cornerRadius(8)

            Text("Password")
                .font(.caption)
                .foregroundStyle(.secondary)
            Text("••••••••")
                .font(.body)
                .padding(.vertical, 8)
                .padding(.horizontal, 12)
                .frame(maxWidth: .infinity)
                .background(Color.gray.opacity(0.1))
                .cornerRadius(8)
        }
    }
    .laneShadowTheme()
    .padding()
}
